manifests/init.pp

```diff
@@ -46,7 +46,10 @@
   $url,
   $destination_directory,
   $destination_file = '',
-  $proxyAddress=''
+  $proxyAddress='',
+  $proxyUser='',
+  $proxyPassword='',
+  $isPasswordSecure=true
```

Comment:
I see how security seems to be a good default, but the implementation here is a bit problematic for Puppet, I feel.

Normally, you want to define node data prior to setting up a machine in the first place, and enable Puppet to make sure everything works coherently. This is inherently incompatible with this mode of password security. As such, I'm inclined to believe that `false` might be a better default here. Users that feel that the proxy credentials must be protected as vehemently as possible would alter this setting. Thoughts?